### ISSUE TITLE
Applying a Filter mechanism fr

### ISSUE THREAD

**Issue by deevodavis71** [mapstruct/mapstruct#1327]

**Comment by sjaakd** [comment]
@deevodavis71 : I don't think that MapStruct should act on any external annotation (such as @jsonview). 

with regards to a solution: Is it possible somehow that you proxy your getter on the source and setters on the target objects  (e.g. by proxying them or use an aspect)?

**Comment by cgendreau** [comment]
The original point 1. (lazily-loaded) could be useful to implement functionalities related to JSON API [sparse-fieldsets ](http://jsonapi.org/format/#fetching-sparse-fieldsets) where we have to return only the fields that have been requested. Checking for `null` on a lazily-loaded property would trigger the loading so it would be handy, in that case, to have a way to check if the property is needed.

Maybe there is alternatives, I don't know.

**Comment by filiphr** [comment]
>Maybe there is alternatives, I don't know.

The only alternative at this moment is to do [Source presence checking](http://mapstruct.org/documentation/stable/reference/html/#source-presence-check) in your objects.
